ui/src/app/memberships/_components/memberships-table-opt-cell.tsx

```diff
@@ -0,0 +1,52 @@
+import { useTransition } from 'react';
+import { Button } from '@/components/ui/button';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faUserPlus, faUserMinus } from '@fortawesome/free-solid-svg-icons';
+import { optIn, optOut } from '@/lib/actions';
+import { useRouter } from 'next/navigation';
+
+const MembershipsOptCell = ({
+    isOptOut,
+    optOutEnabled,
+    groupingPath,
+    removeRow
+}: {
+    isOptOut: boolean;
+    optOutEnabled?: boolean;
+    groupingPath: string;
+    removeRow: (path: string) => void;
+}) => {
+    const [isPending, startTransition] = useTransition();
+    const router = useRouter();
+
+    const handleAction = () => {
+        removeRow(groupingPath);
```

Comment:
Passing around a `removeRow` function looks slightly confusing because it can be any function. What if instead of the `removeRow` variable, we pass the setData (renamed from setTableData) function from the setData. Then the code for removing a row is defined here.

Reply:
Nevermind this regard this.